pkg/apis/pipeline/v1/result_defaults.go

```diff
@@ -21,7 +21,7 @@ func (tr *TaskResult) SetDefaults(context.Context) {
 		return
 	}
 	if tr.Type == "" {
-		if tr.Properties != nil {
+		if tr.Properties != nil && len(tr.Properties) != 0 {
```

Comment:
```suggestion
		if len(tr.Properties) != 0 {
```

Reply:
https://google.github.io/styleguide/go/decisions#nil-slices

Reply:
Learned  ✏️ 📓,  thanks for the pointer Lee!